en/components/grid_column_hiding.md

```diff
@@ -0,0 +1,368 @@
+---
+title: Column Hiding Component – Native Angular | Ignite UI for Angular
+_description: Column hiding with the Ignite UI for Angular Column Hiding component allows the users hide and show columns by using a built-in UI of the Data Grid.
+_keywords: Ignite UI for Angular, UI controls, Angular widgets, web widgets, UI widgets, Angular, Native Angular Components Suite, Native Angular Controls, Native Angular Components Library, Native Angular Components, Angular Data Grid component, Angular Data Grid control, Angular Grid component, Angular Grid control, Angular High Performance Grid, column hiding, hiding, hide
+---
+
+### Grid Column Hiding
+
+The Ignite UI for Angular Data Grid provides a **Column Hiding** component, which allows users to hide and show columns directly through the **UI**. The Data Grid has a built-in column hiding UI, which can be used through the grid's toolbar. In addition, developers can always define the column hiding UI as a separate component and place it anywhere they want on the page.
+
+#### Demo
+
+<div class="sample-container loading" style="height:650px">
+    <iframe id="grid-column-hiding-toolbar-sample-iframe" src='{environment:demosBaseUrl}/grid-column-hiding-toolbar-sample' width="100%" height="100%" seamless frameBorder="0" onload="onSampleIframeContentLoaded(this);"></iframe>
+</div>
+<br/>
+<div>
+<button data-localize="stackblitz" class="stackblitz-btn" data-iframe-id="grid-column-hiding-toolbar-sample-iframe" data-demos-base-url="{environment:demosBaseUrl}">view on stackblitz</button>
+</div>
+<div class="divider--half"></div>
+
+### Grid setup
+Let's start by creating our grid and binding it to our data. We will also enable both filtering and sorting for the columns.
+
+```html
+<!--columnHiding.component.html-->
+
+<igx-grid #grid id="grid" [data]="data" [autoGenerate]="false" width="100%" height="560px" columnWidth="200px"
+            (onColumnInit)="initColumns($event)">
+    <igx-column [field]="'ID'" dataType="string"></igx-column>
+    <igx-column [field]="'ContactName'" dataType="string"></igx-column>
+    <igx-column [field]="'ContactTitle'" dataType="string"></igx-column>
+    <igx-column [field]="'City'" dataType="string"></igx-column>
+    <igx-column [field]="'CompanyName'" dataType="string"></igx-column>
+    <igx-column [field]="'Fax'" dataType="string"></igx-column>
+    <igx-column [field]="'Address'" dataType="string"></igx-column>
+    <igx-column [field]="'PostalCode'" dataType="string"></igx-column>
+    <igx-column [field]="'Country'" dataType="string"></igx-column>
+    <igx-column [field]="'Phone'" dataType="string"></igx-column>
+</igx-grid>
+```
+
+```typescript
+// columnHiding.component.ts
+
+public initColumns(column: IgxColumnComponent) {
```

Comment:
Why not set `filterable` and `sortable` on the column definitions inside the component.html and omit the `initColumns` altogether?